src/writer.rs

```diff
@@ -446,13 +445,21 @@ impl FdtWriter {
     }
 
     /// Write a string property.
-    pub fn property_string(&mut self, name: &str, val: &str) -> Result<()> {
+    pub fn property_string<S: AsRef<str> + Into<Vec<u8>>>(
```

Comment:
I think we could actually just require `S: AsRef<str>` here (and not `Into<Vec<u8>>`) if we change the call below to `CString::new(val.as_ref())`.